src/docmanager/index.ts

```diff
@@ -0,0 +1,438 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+'use strict';
+
+import {
+  IContentsModel, IKernelId, IKernelSpecId, IContentsOpts, IKernel,
+  INotebookSession
+} from 'jupyter-js-services';
+
+import {
+  IDisposable, DisposableDelegate
+} from 'phosphor-disposable';
+
+import {
+  ISignal, Signal
+} from 'phosphor-signaling';
+
+import {
+  Widget
+} from 'phosphor-widget';
+
+
+/**
+ * The interface for a document model.
+ */
+export interface IDocumentModel {
+  /**
+   * A signal emitted when the document content changes.
+   */
+  contentChanged: ISignal<IDocumentModel, any>;
+
+  /**
+   * Serialize the model.  It should return a JSON object or a string.
+   */
+  serialize(): any;
+
+  /**
+   * Deserialize the model from a string or a JSON object.
+   *
+   * #### Notes
+   * Should emit a [contentChanged] signal.
+   */
+  deserialize(value: any): void;
+
+  /**
+   * The default kernel name for the the document.
+   *
+   * #### Notes
+   * This is a read-only property.
+   */
+  defaultKernelName: string;
+
+  /**
+   * The default kernel language for the document.
+   *
+   * #### Notes
+   * This is a read-only property.
+   */
+  defaultKernelLanguage: string;
+}
+
+
+/**
+ * A session info object for running sessions.
+ */
+export interface ISessionInfo {
```

Comment:
Perhaps this should just be
`{path: string; spec: IKernelSpec; kernel: IKernel}` and in `listRunningSessions` we return a list of these things? We do want to return the actual kernel id because the point of having a `listRunningSessions()` function was to be able to see what other kernels are out there so we can connect to the same kernel.